src/services/guild.service.ts

```diff
@@ -1,13 +1,52 @@
 import { ClientService } from './client.service';
-import { Guild } from 'discord.js';
+import { Maybe } from '../common/types';
+import { Guild, GuildChannel, Role } from 'discord.js';
+import Constants from '../common/constants';
 
 export class GuildService {
   private _guild: Guild;
+  private roleCache: Record<string, Maybe<Role>> = {
+    [Constants.Roles.Unacknowledged]: undefined,
+    [Constants.Roles.Unverifed]: undefined,
+  };
+
+  private channelCache: Record<string, Maybe<GuildChannel>> = {
+    [Constants.Channels.Public.CodeOfConduct]: undefined,
+    [Constants.Channels.Bot.Verify]: undefined,
+    [Constants.Channels.Bot.Unacknowledged]: undefined,
+  };
+
   constructor(private _clientService: ClientService) {
     this._guild = this._clientService.guilds.first();
+
+    //Fill role cache
+    Object.keys(this.roleCache).forEach((key) => {
+      if (!this.roleCache[key]) {
+        this.roleCache[key] = this.get()
+          .roles.filter((r) => r.name === key)
+          .first();
+      }
+    });
+
+    //Fill channel cache
+    Object.keys(this.channelCache).forEach((key) => {
+      if (!this.channelCache[key]) {
+        this.channelCache[key] = this.get()
+          .channels.filter((c) => c.name === key)
+          .first();
+      }
+    });
   }
 
   public get() {
     return this._guild;
   }
+
+  public getRole(roleName: string): Role {
+    return this.roleCache[roleName] as Role; //Shouldn't be undefined after constructor
```

Comment:
it can be undefined if the key was never used before. The service should dynamically cache new things as requested.

Reply:
I was going to populate all of them in the constructor. But if we want to do it dynamically, I can remove what I put in the constructor, and just fill 'as needed' in the `getRole/getChannel` functions

Reply:
yep that's a proper cache